### PR TITLE
chore: Update python packaging to use build to produce sdists/wheels

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -57,7 +57,6 @@
     "Jinja2>=2.0.0",
     "jsonschema",
     "mmh3",
-    "numpy<1.22",  # 1.22 drops support for python 3.7.
```
Comment (kevjumba) [inline setup.py:60]: Why remove this?

Comment (achals) [reply setup.py:60]: oh hm I think this was by mistake

File: setup.py [hunk 2]
```diff
@@ -246,17 +247,19 @@ def run(self):
             # so that they are regular packages, and don't need the `--namespace-packages` flags
             # when being typechecked using mypy. BUT, we need to exclude `types` because that clashes
             # with an existing module in the python standard library.
-            if sub_folder == "types":
-                continue
+
+            # if sub_folder == "types":
```
Comment (kevjumba) [inline setup.py:251]: Is this still needed, if not probably should remove.

**Review by kevjumba — APPROVED** [review]

**Pull request merged**